stage/analysis_stage.py

```diff
@@ -27,13 +29,38 @@ class AnalysisStage(BaseStage):
   crashes due to a bug in the project under test or if all major code paths have
   been sufficiently covered."""
 
+  def _analyze_crash(self, result_history: list[Result]) -> Result:
+    """Analyzes a runtime crash."""
+    agent = self.get_agent(agent_name='CrashAnalyzer')
+    if self.args.cloud_experiment_name:
+      return self._execute_agent_cloud(agent, result_history)
+    return agent.execute(result_history)
+
+  def _analyze_coverage(self, result_history: list[Result]) -> None:
+    """Analyzes code coverage."""
+    del result_history
+
   def execute(self, result_history: list[Result]) -> Result:
-    self.logger.info('Analysis Stage')
-    agent = self.get_agent()
-    analysis_result = agent.execute(result_history)
+    """Executes the analysis stage."""
+    last_result = result_history[-1]
+    language = last_result.benchmark.language.lower()
+    if not isinstance(last_result, RunResult):
+      self.logger.error('AnalysisResult must follow a RunResult')
+      raise TypeError
+    semantic_check_result = last_result.semantic_check
+    analysis_result = AnalysisResult(author=repr(self),
+                                     run_result=last_result,
+                                     semantic_result=semantic_check_result)
 
-    # TODO(dongge): Save logs and more info into workdir.
-    self.logger.write_chat_history(analysis_result)
-    self.logger.debug('Analysis stage completed with with result:\n%s',
+    # 1. Analyzing c++/c's runtime crash.
+    if last_result.crashes and language in {'c', 'c++'}:
```

Comment:
Do we need to check language here?
I presume we can have different agents for different languages, but all share the same interface.